src/persist/src/indexed/encoding.rs

```diff
@@ -470,6 +472,26 @@ impl BlobMeta {
                 )));
             }
         }
+
+        let mut batch_keys = HashSet::new();
+        for u in self.unsealeds.iter() {
+            for batch in u.batches.iter() {
+                if batch_keys.contains(&batch.key) {
+                    return Err(format!("duplicate unsealed batch key: {}", batch.key).into());
```

Comment:
super nit: this sounds to me like we disallow duplicates in unsealed (but that a dup with 1 in unsealed and 1 in trace is fine). concretely: I propose `"duplicate batch key found in unsealed: {}"` or something like that

Reply:
fixed here and on trace - thanks!